### ISSUE TITLE
Nginx Not Releasing Deleted Data Files

### ISSUE THREAD

**Issue by stuartio** [51Degrees/Device-Detection#18]
Our nginx.conf has the entry '51D_filePath /etc/51Degrees/51Degrees-EnterpriseV3_2.dat'. We update this file daily and reload Nginx, but it seems the nginx processes don't release the deleted files, many of which can be viewed by running 'lsof | grep deleted'. These files eventually fill our disks unless we restart (rather than reload) the Nginx service. Nginx Plus support also tell me that the the master process should reopen the file on reload, and if not there may be a bug in the module.

**Comment by ben51degrees** [comment]
@stuartio 
Thanks for raising this bug. I have attached a patch which solves this issue as far as I have tested. We will be fully testing this in the next few weeks in order to include it in the next release.

[0001-BUG-Nginx-now-frees-a-previously-allocated-dataset-i.patch.zip](https://github.com/51Degrees/Device-Detection/files/1252158/0001-BUG-Nginx-now-frees-a-previously-allocated-dataset-i.patch.zip)

Regards,
Ben.
